File: bindings/flutter_tessera/macos/flutter_tessera/Sources/flutter_tessera/FlutterTesseraPlugin.swift
// FlutterTesseraPlugin.swift — plugin entry point: registers the platform-view
// factory under the type id the Dart `TesseraView` uses ("flutter_tessera/view").
//
// NOTE: reference implementation for the macOS build (not compiled in this
// environment). See TesseraPlatformView.swift.
import Cocoa
import FlutterMacOS

public class FlutterTesseraPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let factory = TesseraViewFactory(messenger: registrar.messenger)
        registrar.register(factory, withId: "flutter_tessera/view")
    }
}

final class TesseraViewFactory: NSObject, FlutterPlatformViewFactory {
    private let messenger: FlutterBinaryMessenger

    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }

    func create(withViewIdentifier viewId: Int64, arguments args: Any?) -> NSView {
        // Flutter retains the returned NSView, which owns the engine + loop.
        return TesseraPlatformView(viewId: viewId, messenger: messenger, frame: .zero)
    }

    func createArgsCodec() -> (FlutterMessageCodec & NSObjectProtocol)? {
        FlutterStandardMessageCodec.sharedInstance()
    }
}

File: bindings/flutter_tessera/android/src/main/kotlin/com/tessera/flutter_tessera/FlutterTesseraPlugin.kt
package com.tessera.flutter_tessera

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

/** Registers the Tessera platform-view factory ("flutter_tessera/view"). */
class FlutterTesseraPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        binding.platformViewRegistry.registerViewFactory(
            "flutter_tessera/view",
            TesseraViewFactory(binding.binaryMessenger),
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

private class TesseraViewFactory(
    private val messenger: io.flutter.plugin.common.BinaryMessenger,
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: android.content.Context, viewId: Int, args: Any?): PlatformView {
        return TesseraPlatformView(context, viewId, messenger)
    }
}
